### DISCUSSION TITLE
Minimal Hardware Requirement for Loxilb

### DISCUSSION THREAD

**Discussion by jamroks** [loxilb-io/loxilb#624] [Q&A] [answered]
Hi All,

i search through the docs , but couldn't find the answer, sorry if this question has been already answered somewhere, but i would like to know what is : 
- The minimal hardware requirement ?  in terms of  cpu (1vCpu, 2vCpu etc..) and memory (512G, 1G, 2G etc ...)
- The recommended hardware requirement ?  

Thank you

**Comment by jamroks** [comment]
I just tried to launch loxilb on a clean virtual machine with nothing else on it except for ContainerD (which consume even less resources than docker engine ) in AWS which as the following spec : 
- OS : Linux
- Flavor : Ubuntu 22.04
- Platform : ARM64
- ContainerRuntime :  ContainerD (with nerdctl cli)
- Instance Type : T4g.nano 
- CPU : 2vCPUs
- RAM: 512MB

_And when trying to run loxilb docker container like this :_ 

```
sudo nerdctl run -u root --cap-add=SYS_ADMIN --net=host --restart=unless-stopped --privileged -d -t -v /dev/log:/dev/log  --platform=arm64 --name loxilb ghcr.io/loxilb-io/loxilb:latest
```

 _i got the following error_ : 
```
loxilb start
INFO: xdp_link_detach() no curr XDP prog on ifindex:5
21:12:38 DEBUG loxilb_libdp.c:2672: exec: ntc filter del dev llb0 ingress 2>&1 > /dev/null
RTNETLINK answers: Invalid argument
We have an error talking to the kernel
21:12:39 DEBUG loxilb_libdp.c:2733: /opt/loxilb/llb_xdp_main.o: nr 0 psection xdp_packet_hook
libbpf: Error in bpf_create_map_xattr(nh_map):Invalid argument(-22). Retrying without BTF.
libbpf: Error in bpf_create_map_xattr(ct_map):Cannot allocate memory(-12). Retrying without BTF.
libbpf: map 'ct_map': failed to create: Cannot allocate memory(-12)
libbpf: failed to load object '/opt/loxilb/llb_xdp_main.o'
ERR: loading BPF-OBJ file(/opt/loxilb/llb_xdp_main.o) (-12): Cannot allocate memory
ERR: loading file: /opt/loxilb/llb_xdp_main.o
```
As you can see **Cannot allocate memory**  says it all . Please can you advice for a minimal setup,  memory wise ?

**Reply by jamroks** [reply, marked as answer]
allright, the minimal instance type i manage to start loxilb on aws is :
- Instance Type : T4g.micro
- CPU : 2vCPUs
- RAM: 1G
- ContainerRuntime: ContainerD

And i noticed that loxilb used approximately 517.8MiB when idle

**Reply by UltraInstinct14** [reply]
Thanks @jamroks . Loxilb users have reported using it comfortably with t2.medium but it should run fine in instances with lesser resources.
